iphone/Classes/TiUIListView.m

```diff
@@ -2041,74 +2030,52 @@ - (void)searchBarCancelButtonClicked:(UISearchBar *) searchBar
+    if (_searchWrapper != nil) {
+        CGFloat rowWidth = floorf([self computeRowWidth:_tableView]);
+        if (rowWidth > 0) {
+            CGFloat right = _tableView.bounds.size.width - rowWidth;
+#ifndef TI_USE_AUTOLAYOUT
+            [_searchWrapper layoutProperties]->right = TiDimensionDip(right);
+#endif
+             [_searchWrapper refreshView:nil];
+        }
+    }
+    //IOS7 DP3. TableView seems to be adding the searchView to
+    //tableView. Bug on IOS7?
+    [self clearSearchController:self];
+    [_tableView reloadData];
+}
 
-- (void)searchDisplayControllerWillBeginSearch:(UISearchDisplayController *)controller
+- (void)presentSearchController:(UISearchController *)controller
 {
+    id dimBgValue = [[self  proxy] valueForKey:@"dimsBackground"];
+    searchController.dimsBackgroundDuringPresentation = [TiUtils boolValue:dimBgValue def:NO];
+    
     id searchButtonTitle = [searchViewProxy valueForKey:@"cancelButtonTitle"];
     ENSURE_TYPE_OR_NIL(searchButtonTitle, NSString);
     
     if (!searchButtonTitle) {
         return;
     }
-
+    
     // TODO: Use [UIBarButtonItem appearanceWhenContainedInInstancesOfClasses:@[[UISearchBar class]]];
     // as soon as we remove iOS < 9 support
     id searchButton = searchButton = [UIBarButtonItem appearanceWhenContainedIn:[UISearchBar class], nil];
     [searchButton setTitle:[TiUtils stringValue:searchButtonTitle]];
 }
 
--(BOOL)searchDisplayController:(UISearchDisplayController *)controller shouldReloadTableForSearchString:(NSString *)searchString
-{
-    TiColor *resultsBackgroundColor = [TiUtils colorValue:[[self proxy] valueForKey:@"resultsBackgroundColor"]];
-    TiColor * resultsSeparatorColor = [TiUtils colorValue:[[self proxy] valueForKey:@"resultsSeparatorColor"]];
-    id resultsSeparatorInsets = [[self proxy] valueForKey:@"resultsSeparatorInsets"];
-    id resultsSeparatorStyle = [[self proxy] valueForKey:@"resultsSeparatorStyle"];
-    
-    ENSURE_TYPE_OR_NIL(resultsSeparatorInsets, NSDictionary);
-    ENSURE_TYPE_OR_NIL(resultsSeparatorStyle, NSNumber);
-    
-    if (resultsBackgroundColor) {
-        // TIMOB-23281: Hack to support transparent backgrounds (not officially supported)
-        UIColor *color = [resultsBackgroundColor _color] == [UIColor clearColor] ? [UIColor colorWithWhite:1.0 alpha:0.0001] : [resultsBackgroundColor _color];
-        [controller.searchResultsTableView setBackgroundColor:color];
-    }
-    
-    if (resultsSeparatorColor) {
-        [controller.searchResultsTableView setSeparatorColor:[resultsSeparatorColor _color]];
-    }
-    
-    if (resultsSeparatorInsets) {
-        [controller.searchResultsTableView setSeparatorInset:[TiUtils contentInsets:resultsSeparatorInsets]];
-    }
-    
-    if (resultsSeparatorStyle) {
-        [controller.searchResultsTableView setSeparatorStyle:[TiUtils intValue:resultsSeparatorStyle def:UITableViewCellSeparatorStyleSingleLine]];
-    }
-    
-    return YES;
```

Comment:
What happened to this whole block? It was introduced as part of [TIMOB-23281](https://jira.appcelerator.org/browse/TIMOB-23281) and has to stay.